src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs

```diff
@@ -113,16 +112,28 @@ public void Initialize()
         //   in that case we must send the query without the projection
         RewriteQueryForProjection(ref queryTemplate);
 
+        queryTemplates = new(_parent.DatabaseContext.ShardCount);
+        for (int i = 0; i < _parent.DatabaseContext.ShardCount; i++)
+        {
+            queryTemplates.Add(i, queryTemplate.Clone(_context));
```

Comment:
Do we really need to clone it per shard? Reading from blittable should be safe anyway. Do we modify it in the streaming later on?

Reply:
No need. Removed. 
New PR: https://github.com/ravendb/ravendb/pull/14728